// This file is part of the program: Architecture Independent Profiler (AIP).
//
// AIP is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// AIP is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with AIP.  If not, see <http://www.gnu.org/licenses/>.
//
// Authors: Sam Van den Steen, Ghent University
// Copyright: 2016, Ghent University

syntax="proto3";
package utrace_file;

message ID_string{
	// contains string id's I don't want to put in every Histogram message to save space
	repeated string uop_id_to_string = 5;
}

message Stats{
	uint32 inscount = 1;
	uint32 uopcount = 2;
	uint32 loads = 3;
	uint32 stores = 4;
}

message Dependences{
	repeated uint32 rob = 1;
	repeated double	average_path = 2;
	repeated double	branch_path = 3;
	repeated double critical_path = 4;
}

message Histogram{
	repeated uint32 uop_id = 1;
	repeated uint32 uop_freq = 2;
}

message uTrace{
	uint64 utrace_id = 1;
	Stats stats = 2;
	Dependences trace = 3;
	Histogram hist = 4;
}
